js/src/Status/SignerPending/signerPending.js

```diff
@@ -0,0 +1,169 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { Component } from 'react';
+import { observer } from 'mobx-react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import IdentityIcon from '@parity/ui/lib/IdentityIcon';
+import Button from 'semantic-ui-react/dist/commonjs/elements/Button';
+import Container from 'semantic-ui-react/dist/commonjs/elements/Container';
+import Header from 'semantic-ui-react/dist/commonjs/elements/Header';
+import Icon from 'semantic-ui-react/dist/commonjs/elements/Icon';
+import Image from 'semantic-ui-react/dist/commonjs/elements/Image';
+import Label from 'semantic-ui-react/dist/commonjs/elements/Label';
+import List from 'semantic-ui-react/dist/commonjs/elements/List';
+import Popup from 'semantic-ui-react/dist/commonjs/modules/Popup';
+
+import Store from './store';
+import ParityBarStore from '../../ParityBar/store';
+import styles from './signerPending.css';
+
+@observer
+class SignerPending extends Component {
+  static propTypes = {}
+
+  static contextTypes = {
+    api: PropTypes.object.isRequired
+  }
+
+  state = {
+    isOpen: false
+  }
+
+  store = Store.get(this.context.api)
+  parityBarStore = ParityBarStore.get();
+
+  handleViewRequest = () => {
+    this.parityBarStore.toggleOpenSigner();
+    this.handleClose();
+  }
+
+  handleOpen = () => {
+    this.setState({ isOpen: true });
+  }
+
+  handleClose = () => {
+    this.setState({ isOpen: false });
+  }
+
+  renderEtherValue (value) {
```

Comment:
This probably makes sense as a smaller component? (I know we have `render*` methods all-over, but it is a bit of something that says "this really should be a component"